apps/agreement/contracts/Agreement.sol

```diff
@@ -246,20 +247,37 @@ contract Agreement is ILockManager, IAgreement, AragonApp {
     * @notice Update Agreement to title "`_title`" and content "`_content`", with arbitrator `_arbitrator`
     * @dev Initialization check is implicitly provided by the `auth()` modifier
     * @param _arbitrator Address of the IArbitrator that will be used to resolve disputes
-    * @param _aragonAppFeesCashier Cashier contract to handle app transaction fees for new actions
+    * @param _setAppFeesCashier Whether or not the cashier integration should be set up or not
     * @param _title String indicating a short description
     * @param _content Link to a human-readable text that describes the rules for the Agreements instance
     */
     function changeSetting(
         IArbitrator _arbitrator,
-        IAragonAppFeesCashier _aragonAppFeesCashier,
+        bool _setAppFeesCashier,
         string _title,
         bytes _content
     )
         external
         auth(CHANGE_AGREEMENT_ROLE)
     {
-        _newSetting(_arbitrator, _aragonAppFeesCashier, _title, _content);
+        _newSetting(_arbitrator, _setAppFeesCashier, _title, _content);
+    }
+
+    /**
+    * @notice Sync app fees cashier address
+    * @dev The app fees cashier address is being cached in the contract, in case the arbitrator changes the address
+    *      this permission-less function allows anyone to have it synced
+    */
+    function syncAppFeesCashier() external {
```

Comment:
So as it is now, you could:

- Initialize or create a new setting without setting the app cashier
- Then anyone can permissionlessly call `syncAppFeesCashier()` to sync it

While I don't think this behaviour is incorrect—again, I don't see any situation where you would not want to be synced—perhaps this points us to prefer removing the `_setAppFeesCashier` flag entirely?

Reply:
As discussed offline we agreed on having the flag to allow users not to be linked to the Arbitrator's subscription module